docs/devel/vm-logging.md

```diff
@@ -0,0 +1,92 @@
+# VM Logging
+Virtlet runs many VMs on Kubernetes and each of them writes something to `stdout` and `stderr`. User
+is given two options:
+
+1. redirect `stdout`/`stderr` into file (DEFAULT)
+2. redirect `stdout`/`stderr` into pty console
+
+## Redirecting stdout and stderr into file
+By default VM logs are redirected into following file inside Virtlet container:
+```
+/var/log/vms/<sandboxId>/_<retry>.log
+```
+You can verify that this option is enabled by examining `deploy/virtlet-ds.yaml` blueprint. There is
+an option that should specify location where `<sandboxId>` directories should appear:
+```
+- name: VIRTLET_VM_LOG_LOCATION
+  value: "/var/log/vms"
+```
+
+### Architecture
+Following image illustrates how logging is handled. There are two directories on host needed:
+
+* **/var/log/virtlet/vms** (customizable) is where `mirantis/virtlet` container redirects VM stdout
+  into.
+* **/var/log/pods** (predefined by Kubernetes) is where Kubernetes expects JSON formatted logs to appear.
+
+![Logging architecture](../logging-architecture.png)
+
+NOTE: It is important to configure volume mounts for `mirantis/virtlet` and `mirantis/virtlet-log`
+containers according to the scheme above.
+
+### Deploying virtlet-log DaemonSet
+Kubernetes is not able to find and understand raw log files that contain direct dumps from VM. Therefore
+we provide an auxilary container `mirantis/virtlet-log` that must be run as a DaemonSet on each node
+where Virtlet DaemonSet is running. The purpose of this container, whose blueprint is provided in
+`deploy/virtlet-log-ds.yaml`, is reformatting VM logs into a special JSON format that is understood
+by Kubernetes.
+
+Provided that:
+
+* redirecting `stdout` and `stderr` into file is turned on
+* `deploy/virtlet-log-ds.yaml` DaemonSet is deployed and running
+
+user should be able to both see logs on Kubernetes Dashboard:
```

Comment:
nitpick: " to both see [...] but then provide only one option. The other (after the figure) starts a new sentence. Thus, I suggest you have here just `user should be able to see logs on ...`